xds/internal/client/client_xds.go

```diff
@@ -468,6 +481,8 @@ func securityConfigFromCommonTLSContext(common *v3tlspb.CommonTlsContext) (*Secu
 		pi := common.GetValidationContextCertificateProviderInstance()
 		sc.RootInstanceName = pi.GetInstanceName()
 		sc.RootCertName = pi.GetCertificateName()
+	case nil:
+		// It is valid for the validation context to be nil on the server side.
```

Comment:
But won't this fail immediately after return
in `if sc.IdentityInstanceName == ""` on the server side?
in `if sc.RootInstanceName == ""` on the client side?

Reply:
It will fail on the client side because the client side expects some root provider configuration always.

On the server, only the identity provider config is mandatory, and that is set in lines 451-454.

Reply:
Oh, right. It's not the root one.